Gitcoin: Issue #174

Elaniobro opened the pull request:
<!--
  Thank you for your pull request. Please provide a description above and review
  the requirements below.

  Contributors guide: https://github.com/gitcoinco/web/blob/contributing/CONTRIBUTING.md
-->

##### Description
<!-- A description on what this PR aims to solve -->
Coding the design for the iOS landing page as provided in https://github.com/gitcoinco/web/issues/137
* Adding in routing for `ios`
  *  Removing redirect to google form
* Adding ios template
* Adding in routing for `itunes`
  * Stubbed out redirect for `itunes`
* Added in .mp4 video assest

##### Commits that need to be added to this PR:
*  [x] Adding image assets
* [x] Adding ios conditional check(s) for ios specific functionality in `nav.html` and 
`nav_internal.html`
* [x] `/ios` specific css markup
* [x] `ios` template markup
* [x] `href` update app store button urls

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [ ] linter status: 100% pass
 - `npm run stylelint` produces a slew of indentation errors on top of valid stylelint errors. These will not be fixed as part of this PR.
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/contributing/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, ui, crypto, etc). -->
ui

##### Refers/Fixes
#137
#174

Elaniobro commented:
Still work to do. Can we add a `WIP` or `HOLD` label to this PR for now?

thelostone-mc commented:
@Elaniobro let me know when it's done, i'll help review ^_^ I've got some time to spare

Elaniobro commented:
Thanks @thelostone-mc feel free to pull down and let me know your thoughts so far. The CSS has been a little bit of a battle, lots of `!important` that are pulled in that have to be overwritten!

Elaniobro commented:
@thelostone-mc have you had a chance to take a gander?

thelostone-mc commented:
@Elaniobro ah sorry got held up with some work! Checking it now

thelostone-mc commented on app/assets/v2/css/gitcoin.css:
```diff
@@ -315,3 +315,161 @@ div.button-pink {
   color: lightgray;
 }
+
+.ios { 
```
Worth moving this to it's own file rather than keeping it in `gitcoin.css`?

Elaniobro replied:
Yup, was thinking this as well. 

Moved all the ios into its own file.

thelostone-mc commented on app/dashboard/templates/ios.html:
```diff
@@ -0,0 +1,139 @@
+            <div class="row">
+                <div class="col-md-4" style="">
+                    <div class="text-center">
+                        <div class="flat-card">
```
We already have cards set up [here](https://github.com/gitcoinco/web/blob/master/app/dashboard/templates/toolbox.html#L56). Consider reusing them, we could improve them as well ? Just a thought

thelostone-mc commented:
- I see an extra image present behind the video ! 
- the font in the card seems a bit too huge :/ does it look better if it's shrunk down ?
- left a few more comments ^_^
![screenshot-2018-1-6 gitcoin push open source repos forward](https://user-images.githubusercontent.com/5358146/34637651-98d7f292-f26f-11e7-8384-a5e0beec2033.png)

thelostone-mc commented:
(Also not relevant here but @owocki  the navbar has different heights on different page ).
Not sure how this happened 😓 
<img width="1440" alt="screen shot 2018-01-05 at 11 28 06 pm" src="https://user-images.githubusercontent.com/5358146/34637680-396f196a-f270-11e7-825c-18c076aa9912.png">
<img width="1439" alt="screen shot 2018-01-05 at 11 28 17 pm" src="https://user-images.githubusercontent.com/5358146/34637681-398c89d2-f270-11e7-81db-b0a7f30b988b.png">

Elaniobro commented:
@thelostone-mc 
* re-worked the cards to use the existing styles.
* Added 
  * `ios.css`
  * `gitcoin_hover.gif` (transparent version)
    * [ ] - need stactic version of this as well.
* Modified head template to accomadate 
  * `ios.css`
  * `card.css`
  * `toolbox.css`
* Reduced the overall width of the card for mobile
* Made video 100% width on smaller viewports
* Fixed footer padding


Still need the video asset as well as logo asset and href values.

thelostone-mc commented:
If we change the font to Open Sans the text would stay within the card ( @algae12 has to confirm the font )

<img width="1440" alt="screen shot 2018-01-08 at 9 59 02 am" src="https://user-images.githubusercontent.com/5358146/34684615-966b9aee-f45a-11e7-8a67-00721da5b778.png">


Other than that rest looks good 👍 ^_^

Elaniobro commented:
@thelostone-mc odd, thought I had corrected that with `min-height`. 

Thank you for reviewing, if the font is wrong, lets get confirmation on the font to use and I will address both at the same time.

owocki commented:
just got back from my vacation.. and i got say "holy progress batman!". awesome!

lmk if anything needed from me on this :)

Elaniobro commented:
@owocki added in the `ogg` video support, and modified the helmet logo to coincide with the design. I think this is ready to :shipit:  📦  🚀 

Thanks @thelostone-mc for the peer review. 🥂

owocki commented:
reviewing now..

owocki commented:
my first thought upon pulling up `/ios` on my local... "damn, that looks _nice!"

owocki commented:
page looks good in tablet, desktop, and mobile dimensions

owocki commented:
some margin between these two buttons would be nice http://bits.owocki.com/3T172r242p0X/Screen%20Shot%202018-01-09%20at%209.31.59%20PM.png

owocki commented:
on second thought... i dont really see any need for the 'mobile' button in the nav

owocki commented:
just finished looking at the LP.  reviewing code now

owocki commented on README.md:
````diff
@@ -236,8 +236,9 @@ _bountydetails function returns the following fields:
-docker-compose up
+docker-compose up -d
 ```
+Navigate to `http://0.0.0.0:8000/`.
````
LGTM... althought not necessarily related to this PR

Elaniobro replied:
If its ok with you, I'll leave it in. I think this PR will be ready to merge 🌐 in soon.

owocki replied:
ok w me

owocki commented on app/app/sitemap.py:
```diff
@@ -28,6 +28,7 @@ def items(self):
             'whitepaper',
             'whitepaper_access',
             '_leaderboard',
+            'ios',
```
good catch :)

Elaniobro replied:
Danke Shoen! 🎩

owocki commented on app/assets/v2/css/ios.css:
```diff
@@ -0,0 +1,188 @@
+}
+
+.ios .video {
+    width: 100%;
```
inconsistent indentation

Elaniobro replied:
https://github.com/gitcoinco/web/pull/207/commits/fd67299681aed5fc61b9f08a4a608926920e1620 fixes this issue.

owocki commented on app/dashboard/templates/shared/nav_internal.html:
```diff
@@ -18,7 +18,7 @@
         <nav class="navbar navbar-expand-md navbar-dark">
             <a class="navbar-brand pt-0 pb-0" href="/">
-                    <img id="logo" src="/static/v2/images/logo_large.png" data-hover="/static/v2/images/logo_large_hover.gif" width="150" height="50" />
+                    <img id="logo" src="{% if request.path == '/ios' %}/static/v2/images/ios/gitcoin.svg{%else%}/static/v2/images/logo_large.png{% endif %}" data-hover="{% if request.path == '/ios' %}/static/v2/images/ios/gitcoin_hover.gif{%else%}/static/v2/images/logo_large_hover.gif{% endif %}" {% if not request.path == '/ios' %}width="150"{%endif%} height="50" />
```
this line is hard to maintain from my perspective... id rather wrap the whole `<img>` block inside of the `{% if request.path == '/ios' %}` block

Elaniobro replied:
https://github.com/gitcoinco/web/pull/207/commits/72c4d59f1dff1fadb716083a4e5b9a2433fef0bb resolves this.

owocki commented on package-lock.json:
```diff
@@ -31,7 +31,7 @@
     "ansi-styles": {
```
should this file be `.gitignore`'d?

thelostone-mc replied:
Could we leave it here ? 

package-lock ensures that installing modules from the same package.json never results in two different installs for different users! 
The node community kinda encourages folks to not ignore it

Elaniobro replied:
@owocki I made comments on your comments, feel free to leave more comments :)

owocki commented:
i think i will merge this in soon... and just comment out the URL route for now... when apple finally approves the app and we have a release date.. i can uncomment it.

any other changes needed @Elaniobro @thelostone-mc ?

Elaniobro commented:
Hey @owocki which URL do you want commented out specifically? Are we talking `Mobile` in the nav?

owocki commented:
> Are we talking Mobile in the nav?

yes!

Elaniobro commented:
@owocki 
![](https://i.imgflip.com/22guc7.jpg)

thelostone-mc commented:
Yup LGTM !! 
@Elaniobro you might wanna squash the commits :P

Elaniobro commented:
@thelostone-mc how can I? I don't think I can hit the `squash & merge` button....

thelostone-mc commented:
Ah didn't notice the merge commit, 
you could squash the last 23 commits, leave the merge commit as it is and merge the ones before that 

Finally you'd land up with 3 -5 commits, hold on let me give it a shot locally

Elaniobro commented:
@owocki this has been squashed, worked with @thelostone-mc to do so. 👍
